src/deadline_test_fixtures/deadline/worker.py

```diff
@@ -223,10 +225,60 @@ def start(self) -> None:
         self._start_worker_agent()
 
     def stop(self) -> None:
+        worker_id = self.worker_id
         LOG.info(f"Terminating EC2 instance {self.instance_id}")
         self.ec2_client.terminate_instances(InstanceIds=[self.instance_id])
+
+        """
+        When using autoscaling workers are deleted from fleets, 
+        for non autoscaling fleets we need to manually delete the worker.
+        """
+        if not self.configuration.fleet.autoscaling:
+            self.delete_worker_from_fleet(worker_id)
```

Comment:
Two things:

1.  Looks like `worker_id` comes from `self.worker_id`. We can remove the argument to the method and use `self._worker_id` from inside `delete_worker_from_fleet`.
2. I wonder if we should have this call inside the `stop(...)` method or if it makes more sense to make this the responsibility of the `worker` fixture cleanup instead

    https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/8a046adb0c5448107f110c52e72fa73813788eb9/src/deadline_test_fixtures/fixtures.py#L472-L476

Reply:
Changed worker_id to an attribute and set it set it at worker start. delete is now its own method and can be used by the fixture to clean up the worker.